### PR TITLE
LPS-195162 | Test Refactor - Salesforce

### PR DESCRIPTION
**Pull request by victorhcunha**
[LPS-195162](https://liferay.atlassian.net/browse/LPS-195162)

### CODE REVIEW

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
The tests are being passed at ci with these modifications, as you can see here: https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=2428351674&testrayRunId=2428351867

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:250]: Use this pattern "First Name LPS-185429" to the entry value

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:254]: "Last Name LPS-185429"

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:396]: Same here

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:400]: Same here

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:416]: Same here

**Review by JhossephG — CHANGES_REQUESTED** [review]
Good approach here @victorhcunha , just take a look at my comments and remember, avoid create code that does the same of a already existing.

**Comment by JhossephG** [comment]
**More one point:** To resolve this issue definitely we need put in this tearDown file a macro to delete all entries from the salesforce objects used @victorhcunha .

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
I didn't want to add steps to the teardown at this point, this would cause more time to be spent on the CI, my goal here is to try as many new error occurrences using a simple way that doesn't cause as much impact to the test.
But if my modifications do not achieve the expected objective, using teardown would be an alternative.
@JhossephG

**Comment by victorhcunha** [comment]
blocked by https://github.com/liferay-bpm-qa/liferay-portal/pull/1154

**Review by JhossephG — CHANGES_REQUESTED** [review]
You can call this commit message as "Wordsmith in tests using object Contact"

test2

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:118]: Let's remove this section since the original purpose is the test works smoothly with we not needing care about plans B. If in the future the test start to not pass because this, we can update again.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:286]: Same thing as the comment above here

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:375]: Same thing as the comment above here.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:435]: Same thing as the comment above here

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (JhossephG) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:522]: Same thing as the comment above here

**Review by JhossephG — CHANGES_REQUESTED** [review]
Looks good to me, just made these changes that I put on the comments and run the objects suite. Don't forget to attach the index results on the ticket too

**Comment by victorhcunha** [comment]
Right! The requested changes were made, and now the teardown is only enabled to clear the entry after the entry is edited (except for the `AssertCRUDWithFormContainer` test which does not perform this editing step).

**Comment by JhossephG** [comment]
ci:test:object:nocompile

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by JhossephG** [comment]
LGTM! Good job here @victorhcunha, just be sure that the tests are passing on ci using the Objects test suite to see the results and execute locally too. It's ideal before this PR be merged that we clean all the objects that are being used in these tests.

cc: @timpak

**Comment by JhossephG** [comment]
ci:test:object

**Comment by victorhcunha** [comment]
Is blocked by #1189

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase:151]: Why are we setting the var to false here? Shouldn't the tearDown logic only run if it's true? Do we need to explicitly set it to false?

**Review by timpak — CHANGES_REQUESTED** [review]
Hey @victorhcunha, please see my question below. Also, can you change your commit messages to the following?

- `Refactor setUp salesforce tests` 
   - `Extract the configuration logic to the setUp block`
   - The new message clearly describes what you're doing
- `Wordsmith in tests using object Contact` 
   - `Wordsmith`
   - The commit message only needs to be `Wordsmith`, it's a commonly used pattern in Liferay

**Comment by victorhcunha** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase
Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase]: a

Comment (victorhcunha) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase]: It does not necessarily need to be false, it just needs to have any value that is not true. I updated it so that the variable is changed to "", perhaps this will bring better readability.
This change in value in the variable is necessary so that the test does not try to delete an entry that would already be deleted, damaging another round of testing.

**Comment by timpak** [comment]
ci:forward

**Pull request closed without merging**
